sympy/core/numbers.py

```diff
@@ -2842,15 +2842,6 @@ def _eval_evalf(self, prec=None):
     def evalf(self, prec=None, **options):
         return self._eval_evalf(prec)
 
-    @_sympifyit('other', NotImplemented)
-    def __add__(self, other):
-        if isinstance(other, Number):
-            if other is S.NegativeInfinity or other is S.NaN:
```

Comment:
Probably, it would suffice, (as is done in `Number` class)
```py
if isinstance(other, Number) and global_evaluate[0]:
```
I wonder if `evaluate` function is called with other functions having `evaluate` parameter ,contradicting each other, what will happen in such scenario.

Reply:
Like this,
```py
    def __add__(self, other):
        if isinstance(other, Number) and global_evaluate[0]:
            if other is S.NegativeInfinity or other is S.NaN:
                return S.NaN
            return self
        return Number.__add__(self, other)
```
Similarly for other such functions in `Infinity` class.

Reply:
Thanks!